Sources/hostmgr-helper/VirtualMachineSlot.swift

```diff
@@ -63,52 +64,52 @@ class VirtualMachineSlot: NSObject, ObservableObject, VZVirtualMachineDelegate {
         } catch {
             Logger.helper.error("Error launching VM: \(error.localizedDescription)")
             Logger.helper.error("Attempting Cleanup of \(launchConfiguration.handle)")
-            try await vmManager.removeVM(name: launchConfiguration.handle)
 
+            try? await VMManager.removeVM(name: launchConfiguration.handle)
             self.status = .crashed(error)
             throw error
         }
     }
 
     @MainActor
     func stopVirtualMachine() async throws {
-        self.status = .stopping
-        try await virtualMachine?.stop()
-        self.status = .empty
+        switch status {
+        case .starting, .running:
+            self.status = .stopping
+        default:
+            break
+        }
+        try await virtualMachine?.instance.stop()
+        await resetSlot()
     }
 
-    /// Handle a VM that was stopped from outside this object (for instance – by being shut down internally)
+    /// Resets the slot to a stopped status
     ///
     /// No need to do anything except some internal bookkeeping
-    ///
+    /// - Parameter error: Error supplied if the VM crashed
     @MainActor
-    func flush() async throws {
-        self.virtualMachine = nil
-        self.status = .empty
-    }
+    func resetSlot(withError error: Error? = nil) async {
+        if let virtualMachine {
+            try? await VMManager.removeVM(name: virtualMachine.config.handle)
+            self.virtualMachine = nil
+        }
 
-    @MainActor
-    func isRunningVM(withHandle handle: String) -> Bool {
-        switch self.status {
-        case .empty:
-            return false
-        case .starting(let launchConfiguration):
-            return launchConfiguration.handle == handle
-        case .running(let launchConfiguration, _):
-            Logger.helper.debug(
-                "Comparing \(launchConfiguration.handle) and \(handle)"
-            )
-            return launchConfiguration.handle == handle
-        case .stopping:
-            return false
-        case .crashed:
-            return false
+        if let error {
+            self.status = .crashed(error)
+        } else {
+            self.status = .empty
         }
     }
 
     @MainActor
-    func isRunning(virtualMachine: VZVirtualMachine) -> Bool {
-        self.virtualMachine == virtualMachine
+    func isConfiguredForHandle(_ handle: String) -> Bool {
```

Comment:
This was changed to `isConfiguredForHandle` because I think that's the question we really wanted to know. The only caller [was from here](https://github.com/Automattic/hostmgr/blob/c25bd1c8f3cc0e22a95980dd19fad82ebf862c6e/Sources/hostmgr-helper/VMHost.swift#L47), meaning we have the intention to stop it.

In this case no matter what the state I think we proceed with stopping and cleaning up if there's a match.